### PR TITLE
docs: rework issue templates

### PR DESCRIPTION
**Pull request by ephys**
This PR reworks issue templates with the following goals:

- Lead people to github discussions
- Emphasize that failing to provide the necessary information will result in the issue being closed.
- Emphasize that incorrectly formatted code (i.e. not using code blocks) will result in the issue being closed.
- Encourage using https://github.com/sequelize/sequelize-sscce instead of pasting incomplete code.
- Encourage people to use 👍🏼 instead of posting '+1' so we can sort by most reacted issues when prioritizing.
- Rewrite the 'are you willing to open a PR' section: it's not very useful to know that someone who isn't willing to implement the PR is able to do it. The new one conveys the message that they'll either need to do it themselves (with or without help), support financially, or wait until someone else does it
- Add a link to open collective

I know a lot of users will still skip reading any of it and will ignore the template, but we'll just close these issues

### CODE REVIEW

**Review by WikiRik — DISMISSED** [review]
This change does make creating issues less inviting since there is much more focus on OpenCollective but I think it is for the better. Do want to hear what @sdepold thinks as well before we merge this

**Comment by ephys** [comment]
Toning down the last option for "are you willing to open a PR" could be good, as long as users understand that the last checkbox means "I'm willing to wait", it doesn't need to talk about financial support

**Review by fzn0x — APPROVED** [review]

**Review by fzn0x — CHANGES_REQUESTED** [review]
Minor: follow the https://github.com/sequelize/sequelize/blob/main/README.md#book-resource text style

**Comment by ephys** [comment]
@fzn0x they're good changes, I'm not reverting them, but please don't modify branches you don't own - prefer using the PR review suggestion mechanism instead :)

**Comment by fzn0x** [comment]
> @fzn0x they're good changes, I'm not reverting them, but please don't modify branches you don't own - prefer using the PR review suggestion mechanism instead :)

Oops sorry! my bad

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 1]
```diff
@@ -8,54 +8,47 @@ assignees: ""
...
 
 <!-- Remember that first contributors are welcome! -->
 
 - [ ] Yes, I have the time and I know how to start.
-- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
-- [ ] No, I don't have the time, although I believe I could do it if I had the time...
-- [ ] No, I don't have the time and I wouldn't even know how to start.
+- [ ] Yes, I have the time but I will need guidance.
+- [ ] No, I don't have the time, but my company or I are [supporting Sequelize through donations on OpenCollective](https://opencollective.com/sequelize).
+- [ ] No, I don't have the time, I'm not contributing financially, and I understand that I will need to wait until someone from the community or maintainers is interested in implementing my feature.
```
Comment (sdepold) [inline .github/ISSUE_TEMPLATE/feature_request.md:48]: Maybe try to tone down the focus on the finance part

[the remark this answers is not in the document]
Comment (ephys) [reply .github/ISSUE_TEMPLATE/feature_request.md:48]: Yes it sounds good to me

**Review by WikiRik — APPROVED** [review]

**Comment by WikiRik** [comment]
Looks good to me now, @fzn0x do you agree?

**Comment by WikiRik** [comment]
@ephys we forgot to update the other_issue template. do you want to work on that or shall I do that?

**Comment by ephys** [comment]
I don't really know what to do with it. The issue tracker should be for docs improvements, bugs, and feature requests. "other" doesn't fit with that. Best we could do is make it clear that questions belong in discussions / slack / etc and that using that issue type will result in a transfer?

**Comment by WikiRik** [comment]
That's something we could do, but I was mainly thinking of doing the same updates there as we did for the other types. So mainly adding OpenCollective and updating the responses for if people can resolve this issue themselves

**Pull request merged**